CLIENTS.md

```diff
@@ -0,0 +1,266 @@
+# Client Changes for Rekor v2
+
+This document outlines the changes clients need to make to support
+Rekor v2.
+
+## Rekor v2 API
+
+Rekor v2 supports HTTP or gRPC, like Fulcio. For Go, we have implemented a client
+already. For other languages, they can either use the OpenAPI docs or generated
+gRPC service proto, or create their own client.
+
+The service implements one write API, `/api/v2/log/entries`. Example JSON request bodies
+below:
+
+```json
+// Request with a Fulcio certificate
+{
+    "hashedRekordRequestV0_0_2": {
+        "digest": "<base64 digest of artifact>", # Must use hash algorithm from key_details
+        "signature": {
+            "content": "<base64 signature>",
+            "verifier": {
+                "x509Certificate": "<base64 DER-encoded certificate>",
+                "keyDetails": "PKIX_ECDSA_P256_SHA_256" # Must match signing algorithm
+            }
+        }
+    }
+}
+
+// Request with a self-managed key
+{
+    "hashedRekordRequestV0_0_2": {
+        "digest": "<base64 digest of artifact>",
+        "signature": {
+            "content": "<base64 signature>",
+            "verifier": {
+                "publicKey": {
+                    "raw_bytes": "<base64 DER-encoded public key>"
+                },
+                "keyDetails": "PKIX_ECDSA_P256_SHA_256"
+            }
+        }
+    }
+}
+
+// Request with an attestation
+{
+    "dsseRequestV0_0_2": {
+        "envelope": {
+            "payload": "<base64-encoded message>",
+            "payloadType": "<type, e.g. application/vnd.in-toto+json>",
+            "signatures": [
+                {
+                    "sig": "<base64-encoded signature>",
+                    "keyid": ""
+                }
+            ]
+        },
+        "verifier": {
+            "x509Certificate": "<base64 DER-encoded certificate>", # May also be a public key
```

Comment:
Would be clearer if the example was repeated like you are for hashedrekord, with the `publicKey` field instead of `x509Certificate`